hack/ensure-deps/check-docker.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script installs dependencies needed for running build-tce.sh and deploy-tce.sh
+
+BUILD_OS=$(uname -s)
+export BUILD_OS
+# Make sure docker is installed
+echo "Checking for Docker..."
+if [[ -z "$(command -v docker)" ]]; then
+    echo "Installing Docker..."
+    if [[ "$BUILD_OS" == "Linux" ]]; then
+        sudo apt-get update > /dev/null
```

Comment:
Should maybe note somewhere that our e2e tests only support debian based systems?

It may be better to just use the [script based installation](https://docs.docker.com/engine/install/debian/#install-using-the-convenience-script).